bootstrap/messaging/messaging.go

```diff
@@ -0,0 +1,229 @@
+	lc.Infof("Setting options for secure MessageBus with AuthMode='%s' and SecretName='%s",
+		messageBusInfo.AuthMode,
+		messageBusInfo.SecretName)
+
+	secretProvider := container.SecretProviderFrom(dic.Get)
+	if secretProvider == nil {
+		return errors.New("secret provider is missing. Make sure it is specified to be used in bootstrap.Run()")
+	}
+
+	secretData, err := GetSecretData(messageBusInfo.AuthMode, messageBusInfo.SecretName, secretProvider)
+	if err != nil {
+		return fmt.Errorf("Unable to get Secret Data for secure message bus: %w", err)
+	}
+
+	if err := ValidateSecretData(messageBusInfo.AuthMode, messageBusInfo.SecretName, secretData); err != nil {
+		return fmt.Errorf("Secret Data for secure message bus invalid: %w", err)
+	}
+
+	if messageBusInfo.Optional == nil {
+		messageBusInfo.Optional = map[string]string{}
+	}
+
+	// Since already validated, these are the only modes that can be set at this point.
+	switch messageBusInfo.AuthMode {
+	case AuthModeUsernamePassword:
+		messageBusInfo.Optional[OptionsUsernameKey] = secretData.Username
+		messageBusInfo.Optional[OptionsPasswordKey] = secretData.Password
+	case AuthModeCert:
+		messageBusInfo.Optional[OptionsCertPEMBlockKey] = string(secretData.CertPemBlock)
+		messageBusInfo.Optional[OptionsKeyPEMBlockKey] = string(secretData.KeyPemBlock)
+	case AuthModeCA:
+		messageBusInfo.Optional[OptionsCaPEMBlockKey] = string(secretData.CaPemBlock)
+		break
+	}
+
+	return nil
+}
+
+func GetSecretData(authMode string, secretName string, provider SecretDataProvider) (*SecretData, error) {
+	// No Auth? No Problem!...No secrets required.
+	if authMode == AuthModeNone {
+		return nil, nil
+	}
+
+	secrets, err := provider.GetSecret(secretName)
+	if err != nil {
+		return nil, err
+	}
+	data := &SecretData{
+		Username:     secrets[SecretUsernameKey],
+		Password:     secrets[SecretPasswordKey],
+		KeyPemBlock:  []byte(secrets[SecretClientKey]),
+		CertPemBlock: []byte(secrets[SecretClientCert]),
+		CaPemBlock:   []byte(secrets[SecretCACert]),
+	}
+
+	return data, nil
+}
+
+func ValidateSecretData(authMode string, secretName string, secretData *SecretData) error {
+	if authMode == AuthModeUsernamePassword {
+		if secretData.Username == "" || secretData.Password == "" {
+			return fmt.Errorf("AuthModeUsernamePassword selected however Username or Password was not found for secret=%s", secretName)
+		}
+	} else if authMode == AuthModeCert {
+		// need both to make a successful connection
+		if len(secretData.KeyPemBlock) <= 0 || len(secretData.CertPemBlock) <= 0 {
+			return fmt.Errorf("AuthModeCert selected however the key or cert PEM block was not found for secret=%s", secretName)
+		}
+	} else if authMode == AuthModeCA {
+		if len(secretData.CaPemBlock) <= 0 {
+			return fmt.Errorf("AuthModeCA selected however no PEM Block was found for secret=%s", secretName)
+		}
+	} else if authMode != AuthModeNone {
+		return fmt.Errorf("Invalid AuthMode selected")
+	}
```

Comment:
probably using switch-case looks a bit better here;
Also, what happens if none of these else if conditions satisfy, should we add `else` branch at the end just in case?

Reply:
The last else catches it, will look at switch, but I remember Mike originally use else/if for a reason.